### PR TITLE
[Tableau user syncing] Sync multiple domains with one Tableau site

### PR DESCRIPTION
**Pull request by AddisonDunn**
## Product Description

- This PR is an addition to [Tableau User Syncing](https://confluence.dimagi.com/display/USH/Tableau+User+Syncing) that allows syncing the web users of multiple HQ domains with a single Tableau site.
- The code matches domains that have the same [Tableau Server config](https://staging.commcarehq.org/a/addison-test-st/settings/project/integration/tableau_server/) together, and reads all their combined web users as a whole (and associated Tableau user information, roles/groups) when syncing with the Tableau site.
- If the same username exists on more than one of the sites, only one of the users will be synced and an error message is sent to Sentry.

[Jira](https://dimagi-dev.atlassian.net/browse/USH-3051?atlOrigin=eyJpIjoiMzU5YThhMWY4YTVlNDA1MzhlYWNjNDk3YzNmMDYxMzIiLCJwIjoiaiJ9)

## Technical Summary

I think ideally in the future we would create Tableau Server objects that can be selected by multiple domains, which would make this code simpler (we wouldn't have to manually match domains together that have the same tableau server config, i.e. the first commit), but implementing that felt out of scope here.

## Feature Flag

TABLEAU_USER_SYNCING, which is a child toggle of EMBEDDED_TABLEAU.

### Safety story

Tested locally and on staging, and should go through QA. Existing syncing code has been used and tested in the wild.

### Automated test coverage

None exist for the syncing method yet.

### QA Plan

I plan to submit a QA ticket for this, combined with the other Tableau user syncing ticket from this sprint.

### Rollback instructions

- [X] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [X] Risk label is set correctly
- [X] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/reports/util.py [hunk 1]
```diff
@@ -573,15 +573,23 @@ def _get_hq_group_id(session):
 
 @periodic_task(run_every=crontab(minute=0, hour='*/1'), queue='background_queue')
 def sync_all_tableau_users():
+    domains_grouped_by_server = {}  # Looks like {(server name, tableau site): [domains]...}
     for domain in TABLEAU_USER_SYNCING.get_enabled_domains():
-        logger.info(f"Syncing Tableau users on domain: {domain}.")
+        server = TableauServer.objects.get(domain=domain)
+        server_details = (server.server_name, server.target_site)
```
[the remark this answers is not in the document]
Comment (esoergel) [reply corehq/apps/reports/util.py:579]: ah, if it's superuser only then I think we should be okay

**Review by esoergel — APPROVED** [review]

File: corehq/apps/user_importer/tasks.py [hunk 2]
```diff
@@ -116,7 +118,12 @@ def _update_progress(value, start=0):
             update_progress=functools.partial(_update_progress, start=len(group_specs))
         )
         if TABLEAU_USER_SYNCING.enabled(domain):
-            import_tableau_users(domain, user_specs)
+            try:
+                import_tableau_users(domain, user_specs)
+            except TableauAPIError as e:
```
Comment (esoergel) [inline corehq/apps/user_importer/tasks.py:123]: Is there any way to signal this issue to the user as well?  Like can you stick a message in the upload status thingy without causing problems?

**Review by esoergel — APPROVED** [review]

File: corehq/apps/user_importer/tasks.py (hunk 2 above)
Comment (snopoke) [reply corehq/apps/user_importer/tasks.py:123]: Seems like you could append to `user_results["errors"]`

**Review by snopoke — APPROVED** [review]

File: corehq/apps/user_importer/tasks.py (hunk 2 above)
Comment (esoergel) [reply corehq/apps/user_importer/tasks.py:123]: That seems reasonable to me

Comment (esoergel) [reply corehq/apps/user_importer/tasks.py:123]: also, I don't think this needs to block this PR, necessarily

**Pull request merged**